### PR TITLE
Icon badge iOS notifications.

### PR DESCRIPTION
**Pull request by dkorolev**
Hi @mzhurovich 

Icon badge pushes are working.

Depends on: https://github.com/C5T/Current/pull/545

Tested:
- iPad to iOS
- `https://***/admin/data/uid_player_id`
- `https://***/admin/data/push_notifications_marker`

A bit hacky but should do the job for now.

Thanks,
Dima

CC @grixa -- я тут праздную и развлекаюсь, надеюсь, вам тоже не скучно.

### CODE REVIEW

**Comment by dkorolev** [comment]
image

**Comment by dkorolev** [comment]
^ the production code is running all the commits I've pushed above. Good to review & merge.

File: ctfo_server.h [hunk 1]
```diff
@@ -1116,27 +1140,88 @@ class CTFOServer final {
...
+      {"FAV", LOG_EVENT::FAV_CARD},
+      {"FAV_CARD", LOG_EVENT::FAV_CARD},
+      {"UNFAV", LOG_EVENT::UNFAV_CARD},
+      {"UNFAV_CARD", LOG_EVENT::UNFAV_CARD},
+      {"LIKE_COMMENT", LOG_EVENT::LIKE_COMMENT},
+      {"UNLIKE_COMMENT", LOG_EVENT::UNLIKE_COMMENT},
+      {"FLAG_COMMENT", LOG_EVENT::FLAG_COMMENT},
+      {"FLAG_CARD", LOG_EVENT::FLAG_CARD},
+      {"REPORT_USER", LOG_EVENT::REPORT_USER},
+      {"BLOCK_USER", LOG_EVENT::BLOCK_USER},
```
Comment (mzhurovich) [inline ctfo_server.h:1223]: `REPORT_USER` / `BLOCK_USER` - one of them is deprecated, right?

Comment (dkorolev) [inline ctfo_server.h:1223]: Yes as of now, but I don't feel like removing tables from the storage.

File: ctfo_server.h [hunk 2]
```diff
@@ -1116,27 +1140,88 @@ class CTFOServer final {
...
+    using transaction_t = typename Storage::transaction_t;
+
+    PushNotificationsSender(Storage& storage,
+                            std::chrono::microseconds last_pushed_notification_timestamp,
+                            const std::string& onesignal_app_id,
+                            uint16_t onesignal_local_port)
+        : storage(storage),
+          starting_from(last_pushed_notification_timestamp + std::chrono::microseconds(1)),
+          transport(onesignal_app_id, onesignal_local_port) {
+      std::cerr << "Starting sending push notifications from " << starting_from.count() << " epoch us.\n";
```
Comment (mzhurovich) [inline ctfo_server.h:1157]: `DebugPrint`?

Comment (dkorolev) [inline ctfo_server.h:1157]: It's defined in the outer class. OK to remove altogether?

It was here only to debug the logic of when to resume from.

File: ctfo_server.h [hunk 3]
```diff
@@ -1116,27 +1140,88 @@ class CTFOServer final {
...
+                  const auto rhs = fields.uid_player_id[uid];
+                  if (Exists(rhs)) {
+                    return Value(rhs).player_id;
+                  } else {
+                    return "";
+                  }
+                }).Go());
+            if (!player_id.empty()) {
+              // Hack: Update `push_starting_from` on success of one push, leave intact on failure.
+              if (transport.Push(player_id, 1)) {
```
Comment (mzhurovich) [inline ctfo_server.h:1176]: `try / catch`? AFAIR `Push` cant throw?

Comment (dkorolev) [inline ctfo_server.h:1176]: `Push` can throw, but only if things go really bad, in which case I'd like the binary to crash.

**Pull request merged**